server/lorax_server/utils/sources/hub.py

```diff
@@ -28,19 +28,41 @@ def weight_hub_files(
     revision: Optional[str] = None,
     extension: str = ".safetensors",
     api_token: Optional[str] = None,
+    embedding_dim: Optional[int] = None,
 ) -> List[str]:
     """Get the weights filenames on the hub"""
     api = get_hub_api(token=api_token)
     info = api.model_info(model_id, revision=revision)
-    filenames = [
-        s.rfilename
-        for s in info.siblings
-        if s.rfilename.endswith(extension)
-        and len(s.rfilename.split("/")) == 1
-        and "arguments" not in s.rfilename
-        and "args" not in s.rfilename
-        and "training" not in s.rfilename
-    ]
+    if embedding_dim is not None:
+        filenames = [
+            s.rfilename
+            for s in info.siblings
+            if s.rfilename.endswith(extension)
+            and len(s.rfilename.split("/")) <= 2
+            and "arguments" not in s.rfilename
+            and "args" not in s.rfilename
+            and "training" not in s.rfilename
+        ]
+        # Only include the layer for the correct embedding dim
+        embedding_tensor_file = f"2_Dense_{embedding_dim}/model.safetensors"
```

Comment:
Do you know if MRL embeddings always follow the `2_Dense_{embedding_dim}/model.safetensors` file format?

Reply:
for all sentence transformers yes